components/storage/blockdevice/COMPONENT_DATAFLASH/DataFlashBlockDevice.cpp

```diff
@@ -354,6 +354,7 @@ int DataFlashBlockDevice::read(void *buffer, bd_addr_t addr, bd_size_t size)
         result = BD_ERROR_OK;
     }
 
+    _mutex.unlock();
```

Comment:
@geky I couldn't get the first version to work. I browsed through the code and added unlock to all returns that were missing it.

Reply:
Oh wow! Sorry I'm not sure how I missed all of those.